megamek/src/megamek/client/bot/princess/Princess.java

```diff
@@ -172,6 +174,8 @@ public LogLevel getVerbosity() {
     IPathRanker getPathRanker(Entity entity) {
         if(entity.hasETypeFlag(Entity.ETYPE_INFANTRY)) {
             return pathRankers.get(PathRankerType.Infantry);
+        } else if (entity.hasETypeFlag(Entity.ETYPE_AERO) && game.useVectorMove()) {
```

Comment:
This will exclude LAMs in fighter mode. Entity.isAero() includes all aeros and potentially LAMs.